source/blog/2021-03-29-dbal-2.13.md

```diff
@@ -0,0 +1,142 @@
+This release includes additional forward compatibilty to DBAL 3 around
+`Statement` and `Result` API and with an integration in our new
+deprecations logging library.
+
+In addition this DBAL release re-enables PHP 7.1 and 7.2 compatibility to give
+as much flexibility as possible to everyone with forward compatibility.
+
+### Statement and Result Forward Compatibility
+
+DBAL 3.0 extracts all fetch-methods from the `Statement` class and moved them
+to a new `Result` class that is returned from `Statement::execute`. We have
+backported this API to 2.13 - so that you can support writing code for both
+DBAL 2 and 3 at the same time.
+
+Old code:
+
+```php
+$statement = $connection->prepare('SELECT * FROM tbl WHERE col = ?');
+$statement->bindParam(1, $value);
+$statement->execute();
+
+while ($row = $statement->fetch()) {
+}
+
+$connection->executeQuery('SELECT * FROM tbl')->fetchAll();
+```
+
+New Code:
+
+```php
+$statement = $connection->prepare('SELECT * FROM tbl WHERE col = ?');
+$statement->bindParam(1, $value);
+$result = $statement->execute();
+
+while ($row = $result->fetchAssociative()) {
+}
+
+$connection->executeStatement('SELECT * FROM tbl')->fetchAllAssociative();
+```
+
+
+The Result Fetching API was improved to use more human-readable names:
+
+```php
+// Old
+$stmt->fetch();
+$stmt->fetch(PDO::FETCH_ARRAY);
+$stmt->fetchColumn();
+$stmt->fetchAll();
+
+// New
+$result = $stmt->execute();
+$result->fetchAssociative();
+$result->fetchNumeric();
+$result->fetchOne();
+$result->fetchAllAssociative();
+```
+
+Many more changes have been made on the public API and also for the internals,
+but these are the most common ones.
+
+Thank you to [mdumoulin](https://github.com/mdumoulin) for the work on
+improving the forward compatiblity.
+
+### Deprecations Logging
+
+We have wrestled internally for a long time with the strategy on runtime
+deprecation going forward and settled on introducing a tiny abstraction on
+top of `@trigger_error()`.
+
+The reason for this abstraction is the potential for side effects caused by an
+error handler and the potential overhead. We expect our deprecations to be
+triggered a few hundred times in some requests as such the production overhead
+must be minimal. 
+
+This means deprecation logging is disabled by default and you must enable
+it to either use `@trigger_error` or a PSR-3 compatible logger.
+
+See the [deprecation library
+README.md](https://github.com/doctrine/deprecations/) for details on how to
+configure and use it.
+
+### PHP 7.1 and 7.2 Support
+
+A few large Doctrine DBAL deployments still support older versions of PHP that
+are not officially supported anymore, but are covered by support of a few Linux
+distributions.
+
+To provide the largest possible flexibility to the ecosystem to run code with
+both DBAL 2 or 3 this version of Doctrine DBAL will work again with PHP 7.1 and
+7.2
+
+### Migrate to DBAL 3
+
+We recommend a three step strategy to move your code-base and that of your
```

Comment:
Is it "a three-step strategy" or there are two three-step strategies? The existing wording of the intro slightly contradicts the body of the paragraph.